api/src/connection/datasource.ts

```
@@ -0,0 +1,58 @@
+
+/**
+ * Retrieves the connection options for TypeORM DataSource or TypeORMModule.
+ *
+ * @returns The connection options for TypeORM DataSource or TypeORMModule.
+ */
+export const dataSourceOptions = (): DataSourceOptions => {
+    // Safely cast the database type or default to 'mysql'
+    const dbType = (env.TR_DB_TYPE as any) || 'mysql';
+
+    // Parse the port safely with fallback to 3306 if parsing fails
+    const port = isNaN(parseInt(env.TR_DB_PORT, 10)) ? 3306 : parseInt(env.TR_DB_PORT, 10);
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)